proto/v1/issue_service.proto

```diff
@@ -524,17 +524,17 @@ message IssueComment {
   }
 
   message IssueUpdate {
-    optional string from_title = 1;
-    optional string to_title = 2;
-    optional string from_description = 3;
-    optional string to_description = 4;
-    optional IssueStatus from_status = 5;
-    optional IssueStatus to_status = 6;
-    // TODO(d): deprecate from_assignee and to_assignee.
-    // Format: users/{email}
-    optional string from_assignee = 7;
-    // Format: users/{email}
-    optional string to_assignee = 8;
+    enum ChangeType {
+      CHANGE_TYPE_UNSPECIFIED = 0;
+      TITLE = 1;
+      DESCRIPTION = 2;
+      STATUS = 3;
+      LABELS = 4;
+    }
+
+    string from = 1;
```

Comment:
Will take a look all details tomorrow. Quick thought is not changing the existing API if possible.

Reply:
The API would be chaos since we have more and more fields in it..

Reply:
Yup, understood the pain. Figuring out if there's any better way doing it (I need a bit more time). All string types (e.g. status) is my main concern of updated design.

Reply:
I'd like to 1) use original type instead of all strings in case we need i18n or better expression such as repeated string labels, 2) making store and API proto consistent if possible. This is what GPT gives me favoring oneof than optional in our original implementation.

Is it gonna reduce complexity if we employ oneof instead of optional? Instead of combining multiple changes in one issue comment, we can split into multiple issue comment if UpdateIssue() is called with multiple changes.

```
message FieldChange {
    string field_name = 1;

    oneof old_value {
        string old_string_value = 2;
        int32 old_int_value = 3;
        bool old_bool_value = 4;
    }

    oneof new_value {
        string new_string_value = 5;
        int32 new_int_value = 6;
        bool new_bool_value = 7;
    }
}
```

Reply:
LGTM to use `oneof`, but it's hard to change the store to the new message cause we need to DML all issue comments

Reply:
Would you help check if frontend has any call to UpdateIssue() with multiple changes? If not, it should be compatible to change to oneof but have to double check.

```
  message IssueUpdate {
    optional string from_title = 1;
    optional string to_title = 2;
    optional string from_description = 3;
    optional string to_description = 4;
    optional IssueStatus from_status = 5;
    optional IssueStatus to_status = 6;
    // TODO(d): deprecate from_assignee and to_assignee.
    // Format: users/{email}
    optional string from_assignee = 7;
    // Format: users/{email}
    optional string to_assignee = 8;
  }
```
to
```
  message IssueUpdate {
    oneof from_value {
      optional string from_title = 1;
      optional string from_description = 3;
      optional IssueStatus from_status = 5;
    }
    oneof to_value {
      optional string to_title = 2;
      optional string to_description = 4;
      optional IssueStatus to_status = 6;
    }
  }
```

Reply:
Evidence: I changed the issue title. The IssueCommentPayload json does not have the "event" in the tree.

```
{"issueUpdate": {"toTitle": "HelloWorld", "fromTitle": "[2 databases] Change data @07-27 20:44 UTC-0700"}}
```

Reply:
In my mind, the `IssueUpdate` in `IssueComment` is only used to show the changes for the issue (just like the audit log for issues). That's why I'm not very care about the types for `from_value` and `to_value`